sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/metadata/MetadataContextCloseable.java

```diff
@@ -0,0 +1,21 @@
+package org.sdase.commons.server.dropwizard.metadata;
+
+import java.io.Closeable;
+
+/**
+ * A {@link Closeable} that restores a {@link MetadataContext} without throwing an {@link Exception}
+ * when it {@linkplain #close() closes}.
+ */
```

Comment:
Sounds like a stack of context instances. Why do we need to restore the previous context? Don't we always want to start fresh when we process a new HTTP request or Kafka message?

Reply:
We always create a new context when we start processing a new message.

This closable is not related to Kafka. It may be used in other situations as well, e.g. when you restart processing some process that is loaded from the database. That may happen in a request context with it's own metadata context.

Artificial example:

1. Create a claim with the metadata customer-device=claim-landing-page
2. Later add pictures of damage to your claim with the metadata customer-device=app
3. And in the end you have a step in this process where the whole claim is processed with customer-device=customer-portal, e.g. submission to an insurance

In the third step you may trigger some image processing of the files that where uploaded in step 2. You can wrap triggering the image processing to keep the device information of the original upload and the customer-portal context will be restored for the submission in the end.

Reply:
Feels like we should put this into the documentation. 👍